src/utils/Cache.ts

```diff
@@ -0,0 +1,246 @@
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+
+export type CacheTargets = CacheTarget[];
+export enum CacheTarget {
+  CACHE_API = 'CACHE_API',
+  MEMORY = 'MEMORY',
+}
+export type CacheResponse = {
+  data: any;
+  status: number;
+  statusText: string;
+  headers: any;
+};
+
+export const CACHE_API_KEY = 'sentinelhub-v1';
+export const SUPPORTED_TARGETS = [CacheTarget.CACHE_API, CacheTarget.MEMORY];
+const FALLBACK_TARGET = CacheTarget.MEMORY;
+
+export const memoryCache = new Map();
```

Comment:
If we are using `invalidateCaches` in tests, this can be made internal.

Reply:
Alternatively if we are exposing it the function should be documented properly with an example. Don't really know who should decide this. I would say expose it and document it if the users wants to invalidate the cache for some reason (expose more control over caching to the consumer).

Reply:
`memoryCache` is not exported anymore as we are using `invalidateCaches` in the tests now